src/deluge/gui/menu_item/midi/devices.cpp

```diff
@@ -121,26 +123,21 @@ MIDIDevice* Devices::getDevice(int32_t deviceIndex) {
 }
 
 void Devices::drawValue() {
-#if HAVE_OLED
-	renderUIsForOled();
-#else
-	char const* displayName = soundEditor.currentMIDIDevice->getDisplayName();
-	numericDriver.setScrollingText(displayName);
-#endif
+	if (display->type() == DisplayType::OLED) {
+		renderUIsForOled();
+	}
+	else {
+		char const* displayName = soundEditor.currentMIDIDevice->getDisplayName();
+		display->setScrollingText(displayName);
+	}
 }
 
 MenuItem* Devices::selectButtonPress() {
-#if HAVE_OLED
-	midiDeviceMenu.title =
-	    soundEditor.currentMIDIDevice->getDisplayName(); // A bit ugly, but saves us extending a class.
-#endif
 	return &midiDeviceMenu;
 }
 
-#if HAVE_OLED
-
 void Devices::drawPixelsForOled() {
```

Comment:
This menu (the one with multiple devices) also looks distorted on OLED. although when going into a specific device, the device name renders fine as a title.

Reply:
This has been fixed upstream via #344

Reply:
I'll merge that in